.github/workflows/binary.yaml

```diff
@@ -31,6 +31,14 @@ jobs:
       git-commit: ${{ steps.git-commit.outputs.hash }}
     steps:
 
+      - name: Set DATE environment variable
+        run: echo "DATE=$(date +'%Y-%m-%d')" >> "$GITHUB_ENV"
+
+      - name: Set archive environment variables
+        run: |
+          echo "ARCHIVE_TARGZ=dune-$DATE-${{ matrix.name }}.tar.gz" >> $GITHUB_ENV
+          echo "ARCHIVE_DIR=dune-$DATE-${{ matrix.name }}" >> $GITHUB_ENV
```

Comment:
I think you could merge these steps together under _Set environment variables_. Also, what do you think about extracting the ARCHIVE_TARGZ from the ARCHIVE_DIR (`ARCHIVE_TARGZ="$ARCHIVE_DIR.tar.gz" ...`)?

Reply:
The reason I split it up like this is that you can't use a variable set in a step later on in that step. Without the split I wouldn't be able to refer to `DATE` when setting the archive variables. For the same reason, extracting the targz name from the dir name won't work (unless we split them into separate steps of course).

Reply:
Even if you do `DATE=$(date +'%Y-%m-%d')` instead of echoing to the environment?